pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/BrokerCache.java

```diff
@@ -141,18 +143,18 @@ private Map<String, List<BrokerInstance>> getTableToBrokersData() throws Excepti
 
   private BrokerData getBrokerData(Map<String, List<BrokerInstance>> responses) {
     Set<String> brokers = new HashSet<>();
-    Map<String, List<String>> tableToBrokersMap = new HashMap<>();
+    Map<String, List<BrokerInfo>> tableToBrokersMap = new HashMap<>();
```

Comment:
Can you elaborate the reason for moving from String to the BrokerInfo object, is it to fix an existing bug?

Reply:
I made the change to make  `BrokerSelector` more reusable. BrokerSelectors are used in the client, controller and now in Startree's proxy. The network address are generated in different ways. The client and proxy can reuse the ports from the metadata in Helix/ZK. However the controller may have to use the port specified in a different config. `BrokerInfo` maintains the metadata in its canonical form so that module can use the metadata in the way it works for it.

Reply:
Not directly related. This conversation also discusses `BrokerSelector` vs `BrokerInfoSelector`. https://github.com/apache/pinot/pull/13361#discussion_r1660581943